lib/charms/operator_libs_linux/v0/systemd.py

```diff
@@ -0,0 +1,148 @@
+either starts or restarts a service, and service_restart either restarts or starts the
+service.
+
+Example usage:
+```python
+from systemd import *
+
+# Start a service
+if not service_running("mysql"):
+    success = service_start("mysql")
+
+# Attempt to reload a service, restarting if necessary
+success = service_reload("nginx", restart_on_failure=True)
+
+```
+
+"""
+
+import logging
+import subprocess
+
+__all__ = [  # Don't export `service`. (It's not the intended way of using this lib.)
+    "service_pause",
+    "service_reload",
+    "service_restart",
+    "service_resume",
+    "service_running",
+    "service_start",
+    "service_stop",
+]
+
+logger = logging.getLogger(__name__)
+
+
+def service(action: str, service_name: str) -> bool:
+    """Control a system service.
+
+    :param action: the action to take on the service
+    :param service_name: the name of the service to perform th action on
+    """
+    cmd = ["systemctl", action, service_name]
+    if action != "is_active":
+        logger.debug("Attempting to {} '{}' with command {}.".format(action, service_name, cmd))
+    else:
+        logger.debug("Checking if '{}' is active".format(service_name))
+    return subprocess.call(cmd) == 0
+
+
+def service_running(service_name: str) -> bool:
+    """Determine whether a system service is running.
+
+    :param service_name: the name of the service
+    """
+    return service("is-active", service_name)
+
+
+def service_start(service_name: str) -> bool:
+    """Start a system service.
+
+    :param service_name: the name of the service to stop
+    """
+    return service("start", service_name)
+
+
+def service_stop(service_name: str) -> bool:
+    """Stop a system service.
+
+    :param service_name: the name of the service to stop
+    """
+    return service("stop", service_name)
+
+
+def service_restart(service_name: str) -> bool:
+    """Restart a system service.
+
+    :param service_name: the name of the service to restart
+    """
+    return service("restart", service_name)
+
+
+def service_reload(service_name: str, restart_on_failure: bool = False) -> bool:
+    """Reload a system service, optionally falling back to restart if reload fails.
+
+    :param service_name: the name of the service to reload
+    :param restart_on_failure: boolean indicating whether to fallback to a
+                               restart if the reload fails.
+    """
+    service_result = service("reload", service_name)
+    if not service_result and restart_on_failure:
+        service_result = service("restart", service_name)
+    return service_result
+
+
+def service_pause(service_name: str) -> bool:
+    """Pause a system service.
+
+    Stop it, and prevent it from starting again at boot.
+
+    :param service_name: the name of the service to pause
+    """
+    stopped = True
+    if service_running(service_name):
+        stopped = service_stop(service_name)
+    service("disable", service_name)
```

Comment:
`systemctl disable --now` will take care of stopping the service if it is running. This might be simpler.